backend/printcmm.ml

```diff
@@ -220,20 +224,28 @@ let rec expr ppf = function
         el in
       fprintf ppf "@[<1>[%a]@]" tuple el
   | Cop(op, el, dbg) ->
+      with_location_mapping ~label:"Cop" ~dbg ppf (fun () ->
       fprintf ppf "@[<2>(%s" (operation dbg op);
       List.iter (fun e -> fprintf ppf "@ %a" expr e) el;
       begin match op with
       | Capply mty -> fprintf ppf "@ %a" machtype mty
-      | Cextcall { ret; ty_args; alloc = _; name = _; } -> 
+      | Cextcall { ret; ty_args; alloc = _; name = _; } ->
         fprintf ppf "@ %a" extcall_signature (ret, ty_args)
       | _ -> ()
       end;
-      fprintf ppf ")@]"
+      fprintf ppf ")@]")
   | Csequence(e1, e2) ->
       fprintf ppf "@[<2>(seq@ %a@ %a)@]" sequence e1 sequence e2
-  | Cifthenelse(e1, _e2_dbg, e2, _e3_dbg, e3, _dbg) ->
-      fprintf ppf "@[<2>(if@ %a@ %a@ %a)@]" expr e1 expr e2 expr e3
-  | Cswitch(e1, index, cases, _dbg) ->
+  | Cifthenelse(e1, e2_dbg, e2, e3_dbg, e3, dbg) ->
+      with_location_mapping ~label:"Cifthenelse-e1" ~dbg ppf (fun () ->
+      fprintf ppf "@[<2>(if@ %a@ " expr e1;
```

Comment:
```suggestion
      fprintf ppf "@[<2>(if@ %a@ " expr e1);
```

Reply:
I'm not sure, but in that case isn't [dbg] the debuginfo for the whole if? Or only for the condition?

Reply:
I am not sure either, but you are probably right.